dll/win32/shell32/wine/shellpath.c

```diff
@@ -2128,8 +2158,46 @@ HRESULT WINAPI SHGetFolderPathAndSubDirW(
         hr = E_FAIL;
         goto end;
     }
-
     TRACE("Created missing system directory %s\n", debugstr_w(szBuildPath));
+
+    /* create desktop.ini for custom icon */
+    if (CSIDL_Data[folder].nShell32IconIndex)
+    {
+        /* desktop.ini */
+        static const WCHAR s_szDesktopIni[] = { 'd','e','s','k','t','o','p','.','i','n','i',0 };
+        /* %%SystemRoot%%\system32\shell32.dll,%d */
+        static const WCHAR s_szFormat[] = { '%','%','S','y','s','t','e','m','R','o','o','t','%','%','\\',
+            's','y','s','t','e','m','3','2','\\','s','h','e','l','l','3','2','.','d','l','l',',','%','d',0 };
+        /* IconResource */
+        static const WCHAR s_szIconResource[] = { 'I','c','o','n','R','e','s','o','u','r','c','e',0 };
+        /* .ShellClassInfo  */
+        static const WCHAR s_shellClassInfo[] = { '.', 'S', 'h', 'e', 'l', 'l', 'C', 'l', 'a', 's', 's', 'I', 'n', 'f', 'o', 0 };
+
+        WCHAR szIconLocation[MAX_PATH];
+        DWORD dwAttributes;
+
+        /* make the directory a system folder */
+        dwAttributes = GetFileAttributesW(szBuildPath);
+        dwAttributes |= FILE_ATTRIBUTE_SYSTEM;
+        SetFileAttributesW(szBuildPath, dwAttributes);
+
+        /* build the desktop.ini file path */
+        PathAppendW(szBuildPath, s_szDesktopIni);
+
+        /* build the icon location */
+        StringCchPrintfW(szIconLocation, _countof(szIconLocation), s_szFormat,
+                         CSIDL_Data[folder].nShell32IconIndex);
+
+        /* write desktop.ini */
+        WritePrivateProfileStringW(s_shellClassInfo, s_szIconResource, szIconLocation, szBuildPath);
+        WritePrivateProfileStringW(NULL, NULL, NULL, szBuildPath);
```

Comment:
What does this function do when the three first parameters are NULL ?!

Reply:
https://docs.microsoft.com/en-us/windows/desktop/api/winbase/nf-winbase-writeprivateprofilestringa

> After following the preceding steps, an application setup program should call WritePrivateProfileString with the first three parameters set to NULL, and the fourth parameter set to the INI file name. For example: WritePrivateProfileString( NULL, NULL, NULL, L"appname.ini" );
> Such a call causes the mapping of an .ini file to the registry to take effect before the next system reboot. The system rereads the mapping information into shared memory. A user will not have to reboot their computer after installing an application in order to have future invocations of the application see the mapping of the .ini file to the registry.

Reply:
I added a simple comment. OK.